src/ai/defaultai.cc

```diff
@@ -5036,8 +5040,9 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 		if (static_cast<int>(roads.size()) < min_roads_count * (1 + bo.total_count())) {
 			return BuildingNecessity::kForbidden;
 		}
-		bo.primary_priority += (roads.size() - min_roads_count * (1 + bo.total_count())) *
-		                       (2 + std::abs(management_data.get_military_number_at(143)) / 5);
+		bo.primary_priority +=
```

Comment:
here the result is int32_t. The problem in the multiplication is the size() function which is size_type aka size_t. for my understanding the solution might be to simply downcast the size() value to int32_t. as the product will be int32_t due to the total count anyway